rclcpp/include/rclcpp/context.hpp

```diff
@@ -198,27 +213,51 @@ class Context : public std::enable_shared_from_this<Context>
    * \param[in] callback the on shutdown callback to be registered
    * \return the callback passed, for convenience when storing a passed lambda
    */
+  [[deprecated("use add_on_shutdown_callback(OnShutdownCallback) instead")]]
```

Comment:
We might want to keep this out for now, merge this, back port it to galactic, and then make a new pull request that adds this and fixes the warnings on master.